isaric/parsers/ccp-cameroon.toml

```diff
@@ -0,0 +1,828 @@
+    fields = [
+      { field = "q12_signs_symptoms", description = "12) Cough / Toux", values = { 1 = true, 2 = false } },
+      { field = "q13_signs_symptoms", description = "13) Cough - with sputum production / Toux Grasse/ productive", values = { 1 = true, 2 = false } },
+      { field = "q14_signs_symptoms", description = "14) Cough - bloodysputum/haemoptysis / Toux, Hémopthysie/ Crachats sanguinolents", values = { 1 = true, 2 = false } },
+    ]
+
+[[observation]]
+  name = "cough_dry"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+  is_present = { field = "q12_signs_symptoms", description = "12) Cough / Toux", values = { 1 = true, 2 = false } }
+
+[[observation]]
+  name = "cough_with_sputum_production"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q13_signs_symptoms"
+    description = "13) Cough - with sputum production / Toux Grasse/ productive"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "cough_with_haemoptysis"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q14_signs_symptoms"
+    description = "14) Cough - bloodysputum/haemoptysis / Toux, Hémopthysie/ Crachats sanguinolents"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "diarrhoea"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q32_signs_symptoms"
+    description = "32) Diarrhoea / Diarrhée"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "diastolic_blood_pressure_mmHg"
+  phase = "study"
+  start_date = { field = "q2_onset" }
+  date = { field = "q14_treatment_outcome", apply = { function = "isNotNull" } }
+  duration_type = "event"
+  
+  [observation.value]
+    field = "q6_signs_symptoms"
+    source_unit = "mmHg"
+    description = "6) Diastolic BP / Tension artérielle diastolique"
+
+[[observation]]
+  name = "ear_pain"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q17_signs_symptoms"
+    description = "17) Ear pain / Otalgie"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "fatigue_malaise"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q22_signs_symptoms"
+    description = "22) Fatigue , Malaise"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "headache"
+  phase = "admission"
+  date = { ref = "admissionObsDate" }
+
+  [observation.is_present]
+    field = "q27_signs_symptoms"
+    description = "27) Headache / Céphalées"
+    values = { 1 = true, 2 = false }
+
+[[observation]]
+  name = "heart_rate_bpm"
+  phase = "study"
+  start_date = { field = "q2_onset" }
+  date = { field = "q14_treatment_outcome", apply = { function = "isNotNull" } }
+  duration_type = "event"
+  value = { field = "q3_signs_symptoms", description = "3) HR/ Fréquence cardiaque" }
```

Comment:
@ekamau just looking through the complete parser again against my review list - how does this work for observation.heart_rate_bpm? will the value from q3_signs_symptoms only be pulled as a study variable if q14_treatment_outcome is not null? I'm wondering as there are plenty of rows for redcap_event_name = week_2_d14_arm_1 that have no date entered in q14_treatment_outcome and don't want to miss pulling this data. Could you use a condition for the redcap_event_name instead, similar to the pneumonia attributes?

Reply:
@ekamau has this been resolved now? let me know if you'd like to discuss

Reply:
@sadiekelly - I think I have now made the correct changes.
@pipliggins - please have a quick look at the most recent commit and let me know if we need to make further changes to this attribute. thanks.

Reply:
The 'isNotNull' function as used here will result in the date being true/false. If you want to control whether the observation is shown according to the redcap event name, as Sadie suggests, you need to use an 'if' statement to that effect. Removing the apply function and adding the 'if' will mean that the outcome date will be used if provided, _and_ the correct row is being read. If the row is correct but the date is absent, the observation will still be shown but without a date (flagging a warning that the date is missing).

I've corrected this in my edits, have a look and compare to see what I mean.